go/perf/suite/suite.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Package suite implements a performance test suite for Noms, intended for measuring and reporting long running tests.
+//
+// Usage is similar to testify's suite:
+//
+// 1. Define a test suite struct which inherits from `suite.PerfSuite`.
+// 2. Define methods on that struct that start with the word "Test", optionally followed by digits, then followed a non-empty capitalized string.
+// 3. Call `suite.Run` with an instance of that struct.
+// 4. Run `go test` with the `-perf <path to noms db>` flag.
+//
+// Test results are written to Noms, along with a soup of the environment they were recorded in.
+//
+// Test names are derived from that "non-empty capitalized string": `"Test"` is ommitted because it's redundant, and leading digits are omitted to allow for manual test ordering. For example:
```

Comment:
Agree in theory, but here I wanted to be able to measure steps.

* "importing blob" is a useful test in itself, and it the whole import-blob phase was in a Pause clause, it's a waste of perfectly good stats.
* or, we could include both importing blob + parsing CSV in the same perf test, but that muddies what the test is trying to do.
* or, I could make it so that individual perf tests can declare multiple clauses (similar to suite.Pause, have suits.Stage), but this sounded like a PITA to me. I liked that there was a 1:1 correlation between the name of the test, and the measurement that it shows up with.

Besides, these are test suites... arguably, maybe, they're related enough to coordinate ordering amongst themselves.